internal/confmapprovider/discovery/README.md

```diff
@@ -1,6 +1,6 @@
-# Discovery confmap.Provider (Experimental)
+# Discovery confmap.Provider
 
-**This component should not be considered stable. At this time its functionality is provided for testing and validation purposes only.**
+**This feature is currently in [development](https://github.com/open-telemetry/opentelemetry-collector#development)**
```

Comment:
Would the alpha status be more appropriate? Let's discuss the existing state and path to the next phases of maturity.